src/main/java/org/opentripplanner/routing/graph/GraphIndex.java

```diff
@@ -267,15 +269,10 @@ public BitSet servicesRunning (LocalDate date) {
 
         for (TripPattern pattern : patternsForStop.get(stop)) {
 
-            // Use the Lucene PriorityQueue, which has a fixed size
-            PriorityQueue<TripTimeShort> pq = new PriorityQueue<TripTimeShort>(numberOfDepartures) {
-                @Override
-                protected boolean lessThan(TripTimeShort tripTimeShort, TripTimeShort t1) {
-                    // Calculate exact timestamp
-                    return (tripTimeShort.serviceDay + tripTimeShort.realtimeDeparture) >
-                            (t1.serviceDay + t1.realtimeDeparture);
-                }
-            };
+            MinMaxPriorityQueue<TripTimeShort> pq = MinMaxPriorityQueue
+                    .orderedBy(Comparator.comparing((TripTimeShort tts) -> tts.serviceDay + tts.realtimeDeparture))
+                    .maximumSize(numberOfDepartures)
+                    .create();
```

Comment:
The JavaDoc on `MinMaxPrQ` say it is Beta, and also suggest using the Java `PriorityQ` if you do not pull from both ends and use the sizeLimiting functionality. To me it also look like the queue can be replaced with a ArrayList, witch is sorted before getting the values. I know you only wanted to remove the lucene lib, but removing using an array list would make the code easier to understand - or am I missing something?

Reply:
I will add some comment here, after discussing this with @abyrd  and @gmellemstrand.